lambdas/passwordRenewal/src/main/java/uk/gov/di/ipv/cri/drivingpermit/event/handler/PasswordRenewalHandler.java

```diff
@@ -0,0 +1,238 @@
+                                LOGGER.error(
+                                        "Unhandled Exception while handling lambda {} exception {}",
+                                        context.getFunctionName(),
+                                        e.getClass());
+                                LOGGER.error(e.getMessage(), e);
+
+                                eventProbe.counterMetric(
+                                        LAMBDA_DRIVING_PERMIT_CHECK_COMPLETED_ERROR);
+                                throw new RuntimeException(e);
+                            }
+                        });
+        return null;
+    }
+
+    private String generateNewPassword() {
+
+        PasswordGenerator password = new PasswordGenerator();
+
+        CharacterData lowerCaseChars = EnglishCharacterData.LowerCase;
+        CharacterRule lowerCaseRule = new CharacterRule(lowerCaseChars);
+        lowerCaseRule.setNumberOfCharacters(6);
+
+        CharacterData upperCaseChars = EnglishCharacterData.UpperCase;
+        CharacterRule upperCaseRule = new CharacterRule(upperCaseChars);
+        upperCaseRule.setNumberOfCharacters(4);
+
+        CharacterData digitChars = EnglishCharacterData.Digit;
+        CharacterRule digitalRule = new CharacterRule(digitChars);
+        digitalRule.setNumberOfCharacters(2);
+
+        CharacterData specialChars =
+                new CharacterData() {
+                    public String getErrorCode() {
+                        return ERROR_CODE;
+                    }
+
+                    public String getCharacters() {
+                        return "!#$%*()+";
+                    }
+                };
+        CharacterRule splCharRule = new CharacterRule(specialChars);
+        splCharRule.setNumberOfCharacters(2);
+        LOGGER.info("Password has been successfully generated");
+
+        return password.generatePassword(
+                14, splCharRule, lowerCaseRule, upperCaseRule, digitalRule);
+    }
+
+    private void callDVLAApi(String newPassword) throws OAuthErrorResponseException {
+        changePasswordService.sendPasswordChangeRequest(newPassword);
+    }
+
+    private void updateSecret(String secretId, String newPassword) {
+        try {
+            LOGGER.info("Creating a request for Secrets Manager");
+            UpdateSecretRequest secretRequest =
+                    UpdateSecretRequest.builder()
+                            .secretId(secretId)
+                            .secretString(newPassword)
+                            .build();
+            LOGGER.info("Secret Manager Request created");
+
+            secretsManagerClient.updateSecret(secretRequest);
+            LOGGER.info("Secrets Manager Request Sent");
+
+        } catch (SecretsManagerException e) {
+            LOGGER.error(e.awsErrorDetails().errorMessage());
+        }
+    }
+
+    public static String getValue(SecretsManagerClient secretsClient, String secretId) {
```

Comment:
We are only null checking this value in the create step I think we need to be doing this every time its used

Reply:
Should this method be private?